internal/downloader/downloader.go

```diff
@@ -37,7 +43,41 @@ type downloadImpl interface {
 	Download(context.Context, string, []string) error
 }
 
-var gatherFunc = gather.Gather
+var log = logrus.StandardLogger()
+
+var gatherFunc = func(ctx context.Context, source, destination string) (metadata.Metadata, error) {
+	initialize()
+	return gather.Gather(ctx, source, destination)
+}
+
+var _initialize = func() {
+	if log.IsLevelEnabled(logrus.TraceLevel) {
+		goci.Transport = http.NewTracingRoundTripperWithLogger(goci.Transport, log)
```

Comment:
As a follow up, we should use a solution that doesn't rely on global states. Besides being a safer approach it would also remove the need to use `sync`. I understand that that would require changes in go-gather itself. Could you file a follow up issue?